app/services/user_service.rb

```diff
@@ -4,9 +4,27 @@ def self.delete_spam(user)
     user.destroy
   end
 
+  def self.deactivate(user:, actor:, params:)
+    actor.ability.authorize! :deactivate, user
+    UserDeactivationResponse.create(user: user, body: params[:deactivation_response])
+    user.deactivate!
+    user
+  end
+
   def self.update(user:, actor:, params:)
     actor.ability.authorize! :update, user
-    user.update! params
+    user.update params
```

Comment:
We want to nab the validation errors and pass em through, rather than error if a user's not valid